llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1068,7 +1070,15 @@ class VPInstructionWithType : public VPInstruction {
     if (R->isScalarCast())
       return true;
     auto *VPI = dyn_cast<VPInstruction>(R);
-    return VPI && VPI->getOpcode() == VPInstruction::WideIVStep;
+    if (!VPI)
+      return false;
+    switch (VPI->getOpcode()) {
+    case VPInstruction::WideIVStep:
```

Comment:
This patch isn't adding `VPInstruction::WideIVStep`. It feels like this should be done independently in a separate patch?

Reply:
I think the switch is just to match multiple opcodes more easily; not sure if it makes sense to have a separate patch to add a switch for a single opcode first; could also just  add `|| VPI->getOpcode() == VPInstruction::StepVector`

Reply:
Oh sorry, my bad! I didn't spot the existing `VPI->getOpcode() == VPInstruction::WideIVStep`. Apologies for the noise.